model/model.go

```diff
@@ -0,0 +1,68 @@
+package model
+
+type UseCasesInterface interface {
+	Register(request LoginRequest) (*LoginResponse, error)
+	Login(request LoginRequest) (*LoginResponse, error)
+	Logout(token Token) error
+
+	CreateDoc(token Token, doc Doc) (*Doc, error)
```

Comment:
Tokens should be handled by middleware. All usecase functions should operate with some kind of user account, 
specifying user id, roles and access claims, but not JWT!